File: app/models/session.rb
# typed: strict
# frozen_string_literal: true

class Session < ApplicationRecord
  COOKIE_KEY = :mewst_session_token

  has_secure_token

  belongs_to :actor

  sig { params(ip_address: T.nilable(String), user_agent: T.nilable(String), signed_in_at: ActiveSupport::TimeWithZone).returns(Session) }
  def self.start!(ip_address:, user_agent:, signed_in_at: Time.current)
    create!(
      ip_address: ip_address || "",
      user_agent: user_agent || "",
      signed_in_at:
    )
  end
end

File: go/internal/model/session.go
package model

import (
	"time"
)

// Session はセッションのドメインモデル
type Session struct {
	ID         SessionID
	ActorID    ActorID
	Token      string
	IPAddress  string
	UserAgent  string
	SignedInAt time.Time
	CreatedAt  time.Time
	UpdatedAt  time.Time
}
